src/bootstrap/build/compile.rs

```diff
@@ -83,7 +83,7 @@ pub fn std_link(build: &Build,
     add_to_sysroot(&out_dir, &libdir);
 
     if target.contains("musl") &&
-       (target.contains("x86_64") || target.contains("i686")) {
+       (target.contains("x86_64") || target.contains("i686") || target.contains("arm")) {
```

Comment:
Hm perhaps this condition could be reversed to say `if target.contains("musl") && !target.contains("mips")` ?